java/vector/src/main/java/org/apache/arrow/vector/stream/MessageSerializer.java

```diff
@@ -78,12 +78,20 @@ public static int bytesToInt(byte[] bytes) {
    * @throws IOException if something went wrong
    */
   public static long serialize(WriteChannel out, Schema schema) throws IOException {
+    long start = out.getCurrentPosition();
     FlatBufferBuilder builder = new FlatBufferBuilder();
     int schemaOffset = schema.getSchema(builder);
     ByteBuffer serializedMessage = serializeMessage(builder, MessageHeader.Schema, schemaOffset, 0);
-    long size = out.writeIntLittleEndian(serializedMessage.remaining());
-    size += out.write(serializedMessage);
-    return size;
+    int size = serializedMessage.remaining();
+    // ensure that message aligns to 8 byte padding
+    // start position (e.g. magic bytes), 4 bytes for size, message body
+    if ((start + 4 + size) % 8 != 0) {
+      size += 8 - (start + 4 + size) % 8;
+    }
+    out.writeIntLittleEndian(size);
+    out.write(serializedMessage);
+    out.align(); // any bytes written are already captured by our size modification above
+    return size + 4;
```

Comment:
I think currently it might not be a multiple of 8, as we don't pad the magic_bytes, so the align here takes that into account - should we pad the magic bytes as well?